Cubical/ZCohomology/CohomologyRings/CupProductProperties.agda

```diff
@@ -0,0 +1,138 @@
+{-# OPTIONS --safe --experimental-lossy-unification #-}
+module Cubical.ZCohomology.CohomologyRings.CupProductProperties where
+
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.HLevels
+open import Cubical.Foundations.Transport
+
+open import Cubical.Data.Unit
+open import Cubical.Data.Nat hiding (_+_ ; _·_)
+open import Cubical.Data.Int using (ℤ ; pos ; negsuc ; -_ ; _+_ ; _·_ ; +Comm)
+
+open import Cubical.Algebra.Group
+open import Cubical.Algebra.Group.Instances.Unit
+open import Cubical.Algebra.Group.Instances.Int
+open import Cubical.Algebra.Group.Morphisms
+open import Cubical.Algebra.Group.MorphismProperties
+open import Cubical.Algebra.DirectSum.DirectSumHIT.Properties
+
+open import Cubical.HITs.SetTruncation as ST
+
+open import Cubical.ZCohomology.Base
+open import Cubical.ZCohomology.GroupStructure
+open import Cubical.ZCohomology.RingStructure.CupProduct
+open import Cubical.ZCohomology.RingStructure.RingLaws
+
+open PlusBis
+
+
+module _
+  {A : Type ℓ-zero}
+  where
+
+  open SubstLemma ℕ (λ n → coHom n A) (λ n → snd (coHomGroup n A)) public
```

Comment:
Maybe don't open public?

Reply:
but then it has to be open in every single file computing a cohomology ring